spec/Section 3 -- Type System.md

```diff
@@ -1340,14 +1357,281 @@ Union type extensions have the potential to be invalid if incorrectly defined.
+`String` member `a` and a required (non-null) `Int!` member `b`:
+
+```graphql example
+tagged ExampleInputTagged {
+  a: String
+  b: Int!
+}
+```
+
+Literal Value            | Variables               | Coerced Value
+------------------------ | ----------------------- | ---------------------------
+`{ a: "abc", b: 123 }`   | `{}`                    | Error: Exactly one key must be specified
+`{ a: null, b: 123 }`    | `{}`                    | Error: Exactly one key must be specified
+`{ b: 123 }`             | `{}`                    | `{ b: 123 }`
+`{ a: $var, b: 123 }`    | `{ var: null }`         | Error: Exactly one key must be specified
+`{ a: $var, b: 123 }`    | `{}`                    | `{ b: 123 }`
+`{ b: $var }`            | `{ var: 123 }`          | `{ b: 123 }`
+`$var`                   | `{ var: { b: 123 } }`   | `{ b: 123 }`
+`"abc123"`               | `{}`                    | Error: Incorrect value
+`$var`                   | `{ var: "abc123" } }`   | Error: Incorrect value
+`{ a: "abc", b: "123" }` | `{}`                    | Error: Exactly one key must be specified
+`{ b: "123" }`           | `{}`                    | Error: Incorrect value for member {b}
+`{ a: "abc" }`           | `{}`                    | `{ a: "abc" }`
+`{ b: $var }`            | `{}`                    | Error: No keys were specified
+`$var`                   | `{ var: { a: "abc" } }` | `{ a: "abc" }`
+`{ a: "abc", b: null }`  | `{}`                    | Error: Exactly one key must be specified
+`{ b: $var }`            | `{ var: null }`         | Error: {b} must be non-null.
+`{ b: 123, c: "xyz" }`   | `{}`                    | Error: Unexpected member {c}
+
+
+
+**Type Validation**
+
+Tagged types have the potential to be invalid if incorrectly defined. This set
+of rules must be adhered to by every Tagged type in a GraphQL schema.
+
+1. A Tagged type must define one or more members.
+2. For each member of a Tagged type:
+   1. The member must have a unique name within that Tagged type;
+      no two members may share the same name.
+   2. The member must not have a name which begins with the
+      characters {"__"} (two underscores).
+3. {IsOutputType(taggedType)} and {IsInputType(taggedType)} cannot both be {false}.
+
+### Member Deprecation
+
+Members in a Tagged type may be marked as deprecated as deemed necessary by
+the application. It is still legal to query for these members on outputs or
+supply these members in inputs (to ensure existing clients are not broken by
+the change), but the members should be appropriately treated in documentation
+and tooling.
+
+When using the type system definition language, `@deprecated` directives are
+used to indicate that a field is deprecated:
+
+```graphql example
+tagged ExampleType {
+  oldField: String @deprecated
+}
+```
+
+Note: a deprecated member of a tagged type may legitimately be marked as
+non-nullable since this member need not be supplied/returned.
```

Comment:
Not sure I understand this

Reply:
Yeah; not sure how to word that!

For the following tagged type:

```graphql
tagged Foo {
  a: Int!
  b: Int! @deprecated(reason: "Example")
  c: Int!
}
```

Input `{a:1}` is still valid - `b`'s nullability is irrelevant. There's no benefit to enforcing `b` be nullable. If `{b:2}` is supplied it's still valid (despite being deprecated); there's no need to allow `{b: null}` which is (probably) undesirable input.

For output, selection set `{ a b c }` is still valid, and the server can simply choose to never return `b` as a result, so b's nullability is irrelevant.

Is that any clearer? How can I make this understandable. (Namely the behaviour of @deprecated w.r.t nullability differs to the rest of the spec.)

Reply:
I've pushed a slightly reworded note.